project-koku/koku
# [COST-1802]-[COST-1806] AWS savings plan

### Pull request for COST-1802 through COST-1806
* [COST-1802:](https://issues.redhat.com/browse/COST-1802) Require savingsplan effective cost in trino
* [COST-1803](https://issues.redhat.com/browse/COST-1803): Migrate existing aws trino tables to force inclusion of savings plan
* [COST-1804](https://issues.redhat.com/browse/COST-1804): add savings plan column to parquet report processor
* [COST-1805](https://issues.redhat.com/browse/COST-1805): Add savings plan to django models
* [COST-1806](https://issues.redhat.com/browse/COST-1806): populate savings plan cost 
### Testing 
1. Checkout the branch and bring everything up with trino: 
```
make docker-up-min-presto
```
2. Create a test customer and load data: 
```
make create-test-customer
make load-test-customer-data
```
3. Once that processes, make sure the `savingsplan_effective_cost` is populated: 
for line item daily
```
PGPASSWORD=postgres psql postgres -U postgres -h localhost -p 15432
SELECT * FROM acct10001.reporting_awscostentrylineitem_daily_summary;
```
for presto
```
presto --server localhost:8080 --catalog hive --schema acct1001 --user admin --debug
SELECT * FROM aws_line_items LIMIT 1;
SELECT * FROM aws_line_items_daily LIMIT 1;
SELECT * FROM aws_openshift_daily LIMIT 1;
```
### Testing the ClowdJobInvocation 
Deploy to the ephemeral cluster 
Manually create a source 
Apply the CJI
```
oc apply -f deploy/koku-script-runner-job.yaml
```
Check the pod logs 
Check the trino table 

### Local smoke tests results
```
===================================================== short test summary info =====================================================
FAILED lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test_source.py::test_api_aws_source_raw_calc - Assert...
FAILED lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test_source.py::test_api_aws_source_raw_calc_second_source_cost_model
FAILED lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test_source.py::test_api_multi_ingest_ocp_source - As...
FAILED lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test_source.py::test_api_multiple_aws_sources_ingest
FAILED lib64/python3.6/site-packages/iqe_cost_management/tests/rest_api/v1/test_source.py::test_api_multiple_ocp_on_aws_sources_ingest
============================ 5 failed, 5809 passed, 196 skipped, 618 deselected in 4618.68s (1:16:58) =============================
```
